src/bun.js/test/expect.zig

```diff
@@ -221,44 +221,38 @@ pub const Expect = struct {
     pub fn getSnapshotName(this: *Expect, allocator: std.mem.Allocator, hint: string) ![]const u8 {
         const test_name = this.scope.tests.items[this.test_id].label;
 
-        var length: usize = 0;
+        var builder = bun.StringBuilder{};
+
         var curr_scope: ?*DescribeScope = this.scope;
         while (curr_scope) |scope| {
             if (scope.label.len > 0) {
-                length += scope.label.len + 1;
+                builder.count(scope.label);
+                builder.count(" ");
             }
             curr_scope = scope.parent;
         }
-        length += test_name.len;
+        builder.count(test_name);
         if (hint.len > 0) {
-            length += hint.len + 2;
+            builder.count(hint);
+            builder.count(": ");
         }
 
-        var buf = try allocator.alloc(u8, length);
+        try builder.allocate(allocator);
 
-        var index = buf.len;
-        if (hint.len > 0) {
-            index -= hint.len;
-            bun.copy(u8, buf[index..], hint);
-            index -= test_name.len + 2;
-            bun.copy(u8, buf[index..], test_name);
-            bun.copy(u8, buf[index + test_name.len ..], ": ");
-        } else {
-            index -= test_name.len;
-            bun.copy(u8, buf[index..], test_name);
-        }
-        // copy describe scopes in reverse order
-        curr_scope = this.scope;
         while (curr_scope) |scope| {
             if (scope.label.len > 0) {
-                index -= scope.label.len + 1;
-                bun.copy(u8, buf[index..], scope.label);
-                buf[index + scope.label.len] = ' ';
+                builder.write(scope.label);
```

Comment:
this will copy describe scope names in child -> parent order. It needs to go parent -> child